src/test/java/com/google/api/codegen/testdata/go_main_library.baseline

```diff
@@ -48,32 +80,23 @@ func defaultClientSettings() gax.ClientSettings {
         gax.WithDelayTimeoutSettings(100*time.Millisecond, 1000*time.Millisecond, 1.2),
         gax.WithRPCTimeoutSettings(300*time.Millisecond, 3000*time.Millisecond, 1.3),
     }
-    return gax.ClientSettings{
-        AppName: "gax",
-        AppVersion: gax.Version,
-        Endpoint: "library-example.googleapis.com:443",
-        Scopes: []string{
-            "https://www.googleapis.com/auth/library",
-            "https://www.googleapis.com/auth/cloud-platform",
-        },
-        CallOptions: map[string][]gax.CallOption{
-            "CreateShelf": defaultRetryOptions,
-            "GetShelf": append([]gax.CallOption{withIdempotentRetryCodes}, defaultRetryOptions...),
-            "ListShelves": append([]gax.CallOption{withIdempotentRetryCodes}, defaultRetryOptions...),
-            "DeleteShelf": append([]gax.CallOption{withIdempotentRetryCodes}, defaultRetryOptions...),
-            "MergeShelves": defaultRetryOptions,
-            "CreateBook": defaultRetryOptions,
-            "PublishSeries": defaultRetryOptions,
-            "GetBook": append([]gax.CallOption{withIdempotentRetryCodes}, defaultRetryOptions...),
-            "ListBooks": append([]gax.CallOption{withIdempotentRetryCodes}, defaultRetryOptions...),
-            "DeleteBook": append([]gax.CallOption{withIdempotentRetryCodes}, defaultRetryOptions...),
-            "UpdateBook": append([]gax.CallOption{withIdempotentRetryCodes}, defaultRetryOptions...),
-            "MoveBook": defaultRetryOptions,
-            "ListStrings": append([]gax.CallOption{withIdempotentRetryCodes}, defaultRetryOptions...),
-            "AddComments": defaultRetryOptions,
-            "GetBookFromArchive": append([]gax.CallOption{withIdempotentRetryCodes}, defaultRetryOptions...),
-            "UpdateBookIndex": append([]gax.CallOption{withIdempotentRetryCodes}, defaultRetryOptions...),
-        },
+    return &CallOptions{
+        CreateShelf: defaultRetryOptions,
+        GetShelf: append(defaultRetryOptions, withIdempotentRetryCodes),
```

Comment:
This can modify defaultRetryOptions, so maybe make that a function as with defaultCallOptions.